### ISSUE TITLE
Speed issue during nncf.quantize_with_accuracy_control

### ISSUE THREAD

**Issue by CengizhanYurdakul** [openvinotoolkit/nncf#1652]
When quantizing my model with `nncf.quantize_with_accuracy_control`, the process is very slow even though I am using a single image. Is it possible to set up an environment or parameter where I can do this faster ?

**Comment by MaximProshin** [comment]
@alexsu52 , @andrey-churkin , please take a look.

**Comment by andrey-churkin** [comment]
@CengizhanYurdakul Hello! Thanks for your question! Could you please give me more details? Could you please provide a script so I can reproduce the problem? Which model do you use?

**Comment by CengizhanYurdakul** [comment]
@andrey-churkin Hello,
I am trying to convert a generator from GANs. I cannot share the model with you for now, but I can show the quantisation codes as follows.
```
# Validation method
def validate(model: openvino.runtime.CompiledModel, validation_loader: torch.utils.data.DataLoader):
   # Caculate MSE between ground truth and generated
    return MSE

# Load model
core = Core()
model = core.read_model("Generator.xml")

# Load dataset
dataset = CustomDataset("./Images", 128)
def transformFunction(data_item):
    return data_item

quantizationDataset = nncf.Dataset(dataset, transformFunction)
validationDataset = nncf.Dataset(dataset, transformFunction)

# Quantization
quantizedModel = nncf.quantize_with_accuracy_control(model,
                                    calibration_dataset=quantizationDataset,
                                    validation_dataset=validationDataset,
                                    validation_fn=validate,
                                    max_drop=0.003)
```
Generator is running 5-6 FPS on the CPU. In my opinion, there are too many iterations on the CPU during the `validate` method, so a lot of time is spent during updates. Actually what I want to ask is if there is a way to speed up this process (like doing it on GPU)?

**Comment by alexsu52** [comment]
Hi @CengizhanYurdakul,

`quantize_with_accuracy_control` method require to collect metric per sample from validation dataset for ranging quantized operations to find operation with highest quantization error that will be reverted to FP32 precision. It's required double pass through dataset: the first pass for calculation full metric and second pass to calculate per sample metric. 

The following PR https://github.com/openvinotoolkit/nncf/pull/1809 extend API of validation function to join calculation full metric and per sample metric in one pass through dataset. You need add calculation per sample metric in the your validation function to speed up `quantize_with_accuracy_control` like this:
```
# Validation method
def validate(model: openvino.runtime.CompiledModel, validation_loader: torch.utils.data.DataLoader):
   # Caculate MSE between ground truth and generated
    return MSE, List[MSE] # return per-sample MSE
```
